packages/edit-site/src/components/sidebar-edit-mode/page-panels/page-status.js

```diff
@@ -0,0 +1,229 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	Button,
+	BaseControl,
+	ToggleControl,
+	Dropdown,
+	__experimentalText as Text,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	TextControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useDispatch } from '@wordpress/data';
+import { useState, useMemo } from '@wordpress/element';
+import { store as coreStore } from '@wordpress/core-data';
+import { store as noticesStore } from '@wordpress/notices';
+import { __experimentalInspectorPopoverHeader as InspectorPopoverHeader } from '@wordpress/block-editor';
+/**
+ * Internal dependencies
+ */
+import StatusLabel from '../../sidebar-navigation-screen-page/status-label';
+
+const STATUS_OPTIONS = [
+	{
+		label: __( 'Draft' ),
+		value: 'draft',
+		hint: __( 'Not ready to publish.' ),
+	},
+	{
+		label: __( 'Pending' ),
+		value: 'pending',
+		hint: __( 'Waiting for review before publishing.' ),
+	},
+	{
+		label: __( 'Private' ),
+		value: 'private',
+		hint: __( 'Only visible to site admins and editors.' ),
+	},
+	{
+		label: __( 'Scheduled' ),
+		value: 'future',
+		hint: __( 'Publish automatically on a chosen date.' ),
+	},
+	{
+		label: __( 'Published' ),
+		value: 'publish',
+		hint: __( 'Visible to everyone.' ),
+	},
+];
+
+export default function PageStatus( {
+	postType,
+	postId,
+	status,
+	password,
+	date,
+} ) {
+	const [ showPassword, setShowPassword ] = useState( !! password );
+
+	const { editEntityRecord } = useDispatch( coreStore );
+	const { createErrorNotice } = useDispatch( noticesStore );
+
+	const [ popoverAnchor, setPopoverAnchor ] = useState( null );
+	// Memoize popoverProps to avoid returning a new object every time.
+	const popoverProps = useMemo(
+		() => ( {
+			// Anchor the popover to the middle of the entire row so that it doesn't
+			// move around when the label changes.
+			anchor: popoverAnchor,
+			placement: 'bottom-end',
```

Comment:
The popover doesn't have a descriptive name for screen readers, users might not even know that they land in a popover. Maybe we should add `aria-label` or `aria-labelledby` here.

The other thing is that `Popover` traps the focus but it doesn't set the `dialog` role. This could be a bigger problem in general of the component design and I personally think it's an accessibility bug. We could set a `role: 'dialog'` here but I'm not sure if it's the right move. Possibly better to consult an accessibility expert on this.

It's still usable thought just a bit confusing so I don't think this should be blocking 👍

Reply:
Added an `aria-label` for the time being since I don't want to modify `InspectorPopoverHeader` or `PublishDateTimePicker` as part of this PR to add ID to title. 

> The other thing is that Popover traps the focus but it doesn't set the dialog role. This could be a bigger problem in general of the component design and I personally think it's an accessibility bug. We could set a role: 'dialog' here but I'm not sure if it's the right move. Possibly better to consult an accessibility expert on this.

Let's create a separate issue referencing the Dropdown and Popover components in wordpress/components to review any accessibility concerns. [Modal](https://wordpress.github.io/gutenberg/?path=/story/components-modal--default) has a `title` prop that uses `aria-labelledby`.